Add endpoint to remove web push subscription

oneiros opened the pull request:
... and include in an HTTP header to be used by the receiving party in case of delivery errors.

Fixes MAS-328

oneiros committed:
Add endpoint to remove web push subscription

and include in an HTTP header to be used by the receiving
party in case of delivery errors.

renchap commented on app/controllers/api/web/push_subscriptions_controller.rb:
```diff
@@ -17,6 +17,13 @@ def update
   end
 
+  def destroy
+    push_subscription = ::Web::PushSubscription.find_by_token_for(:unsubscribe, params[:id])
```
What happens if we cant find this subscription? I think it should be a no-op?

oneiros replied:
Good point. This line will actually not fail, but the next would raise an error. I fixed that just now.

oneiros committed:
Ignore missing subscription or invalid token

oneiros committed:
Re-use TTL constant.

This was the reason for the 48 hour expiration and this change
makes this obvious _and_ ensures these values stay in sync.

ThisIsMissEm commented on spec/requests/api/web/push_subscriptions_spec.rb:
```diff
@@ -0,0 +1,21 @@
+      web_push_subscription.generate_token_for(:unsubscribe)
+    end
+
+    it 'deletes the subscription' do
+      expect { subject }
+        .to change(Web::PushSubscription, :count).by(-1)
+    end
```
Probably needs a test case for the "subscription not found" case that renaud pointed out, just to assert it definitely always gives a 200 head response

oneiros replied:
Good point. I just added two more test cases.

ThisIsMissEm replied:
This seems good to me now, I'll leave it up to y'all as to when/if this gets merged, but implementation wise this looks solid to me.

ThisIsMissEm reviewed (commented):
This looks good to me, based on what Renaud told me about this, only concern is that extra test case.

oneiros committed:
Add more test cases
